content/src/Server.ts

```diff
@@ -47,12 +50,18 @@ export class Server {
     const upload = multer({ dest: Server.UPLOADS_DIRECTORY, preservePath: true })
     const controller: Controller = env.getBean(Bean.CONTROLLER)
     this.garbageCollectionManager = env.getBean(Bean.GARBAGE_COLLECTION_MANAGER)
-    this.synchronizationManager = env.getBean(Bean.SYNCHRONIZATION_MANAGER)
     this.snapshotManager = env.getBean(Bean.SNAPSHOT_MANAGER)
     this.service = env.getBean(Bean.SERVICE)
     this.migrationManager = env.getBean(Bean.MIGRATION_MANAGER)
     this.repository = env.getBean(Bean.REPOSITORY)
 
+    if (env.getConfig(EnvironmentConfig.DISABLE_SYNCHRONIZATION)) {
```

Comment:
Let's move this complexity inside the `SynchronizationManager`. We don't want to have all domain config & logic be here in one place.

We can move this inside the `SynchronizationManager#start`